go/vt/vtadmin/cluster/cluster.go

```diff
@@ -373,50 +374,269 @@ func (c *Cluster) getTablets(ctx context.Context) ([]*vtadminpb.Tablet, error) {
+	span.Annotate("keyspace", keyspace)
+	annotateGetSchemaRequest(opts.BaseRequest, span)
+	vtadminproto.AnnotateSpanWithGetSchemaTableSizeOptions(opts.SizeOpts, span)
 
-	// Copy the request to not mutate the caller's request object.
-	r := *req
-	r.TabletAlias = tablet.Tablet.Alias
+	if len(opts.Tablets) == 0 {
+		// Fetch all tablets for the keyspace.
+		var err error
 
-	span.Annotate("tablet_alias", topoproto.TabletAliasString(r.TabletAlias))
-	span.Annotate("exclude_tables", strings.Join(r.ExcludeTables, ","))
-	span.Annotate("tables", strings.Join(r.Tables, ","))
-	span.Annotate("include_views", r.IncludeViews)
-	span.Annotate("table_names_only", r.TableNamesOnly)
-	span.Annotate("table_sizes_only", r.TableSizesOnly)
+		opts.Tablets, err = c.FindTablets(ctx, func(tablet *vtadminpb.Tablet) bool {
+			return tablet.Tablet.Keyspace == keyspace
+		}, -1)
+		if err != nil {
+			return nil, fmt.Errorf("%w for keyspace %s", errors.ErrNoTablet, keyspace)
+		}
+	}
 
-	schema, err := c.Vtctld.GetSchema(ctx, &r)
-	if err != nil {
-		return nil, err
+	if err := c.Vtctld.Dial(ctx); err != nil {
+		return nil, fmt.Errorf("failed to Dial vtctld for cluster = %s for GetSchema: %w", c.ID, err)
 	}
 
-	if schema == nil || schema.Schema == nil {
-		return nil, nil
+	var tabletsToQuery []*vtadminpb.Tablet
+
+	if opts.SizeOpts.AggregateSizes {
+		span, ctx := trace.NewSpan(ctx, "Cluster.FindAllShardsInKeyspace")
+
+		AnnotateSpan(c, span)
+		span.Annotate("keyspace", keyspace)
+
+		resp, err := c.Vtctld.FindAllShardsInKeyspace(ctx, &vtctldatapb.FindAllShardsInKeyspaceRequest{
+			Keyspace: keyspace,
+		})
+
+		span.Finish()
+
+		if err != nil {
+			return nil, fmt.Errorf("FindAllShardsInKeyspace(cluster = %s, keyspace = %s) failed: %w", c.ID, keyspace, err)
+		}
+
+		for _, shard := range resp.Shards {
+			if !shard.Shard.IsMasterServing {
+				if !opts.SizeOpts.IncludeNonServingShards {
+					log.Infof("%s/%s is not serving; ignoring because IncludeNonServingShards=false", keyspace, shard.Name)
+					continue
+				}
+			}
+
+			shardTablets := vtadminproto.FilterTablets(func(tablet *vtadminpb.Tablet) bool {
+				return tablet.Tablet.Shard == shard.Name && tablet.State == vtadminpb.Tablet_SERVING
+			}, opts.Tablets, len(opts.Tablets))
+
+			if len(shardTablets) == 0 {
+				return nil, fmt.Errorf("%w for shard %s/%s", errors.ErrNoServingTablet, shard.Keyspace, shard.Name)
+			}
+
+			randomServingTablet := shardTablets[rand.Intn(len(shardTablets))]
+			tabletsToQuery = append(tabletsToQuery, randomServingTablet)
+		}
+	} else {
+		keyspaceTablets := vtadminproto.FilterTablets(func(tablet *vtadminpb.Tablet) bool {
+			return tablet.Tablet.Keyspace == keyspace && tablet.State == vtadminpb.Tablet_SERVING
+		}, opts.Tablets, len(opts.Tablets))
+
+		if len(keyspaceTablets) == 0 {
+			// consider how to include info about the tablets we looked at, but
+			// that's also potentially a very long list .... maybe we should
+			// just log it (yes, do that).
```

Comment:
Should this be a formal `TODO`? Haha 🌻

Reply:
I uhh meant for this to be for my eyes only 😬